detekt/detekt_test.bzl

```diff
@@ -0,0 +1,83 @@
+"""
+The rule unit tests.
+"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+load(":detekt.bzl", "detekt")
+
+def _detekt_action_contents_test_impl(ctx):
+    expected_arguments = [
+        "java",
+        "-Xms16m",
+        "-Xmx128m",
+        "-cp",
+        "external/detekt_cli_jar/file/downloaded:bazel-out/darwin-fastbuild/bin/detekt/wrapper/bin_deploy.jar",
+        "io.buildfoundation.bazel.rulesdetekt.wrapper.Main",
+        "--config",
+        "detekt/config.yml",
+        "--input",
+        "detekt/path A",
+        "detekt/path B",
+        "detekt/path C",
+        "--parallel",
+        "--report",
+        "txt:bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.txt",
+        "--report",
+        "xml:bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.xml",
+        "--report",
+        "html:bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.html",
+    ]
+
+    expected_inputs = [
+        "detekt/path A",
+        "detekt/path B",
+        "detekt/path C",
+        "detekt/config.yml",
+        "bazel-out/darwin-fastbuild/bin/detekt/wrapper/bin_deploy.jar",
+        "external/detekt_cli_jar/file/downloaded",
+    ]
+
+    expected_outputs = [
+        "bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.txt",
+        "bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.xml",
+        "bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.html",
+    ]
+
+    env = analysistest.begin(ctx)
+
+    actions = analysistest.target_actions(env)
+
+    asserts.equals(env, 1, len(actions))
+
+    action = actions[0]
+
+    asserts.equals(env, expected_arguments, action.argv)
+    asserts.equals(env, expected_inputs, [file.path for file in action.inputs.to_list()])
+    asserts.equals(env, expected_outputs, [file.path for file in action.outputs.to_list()])
+
+    return analysistest.end(env)
+
+detekt_action_contents_test = analysistest.make(_detekt_action_contents_test_impl)
+
+def _test_detekt_action_contents():
+    detekt(
+        name = "test_target",
+        srcs = ["path A", "path B", "path C"],
+        config = "config.yml",
+        parallel = True,
+        html_report = True,
+        xml_report = True,
+    )
+
+    detekt_action_contents_test(
+        name = "detekt_action_contents_test",
+        target_under_test = ":test_target",
+    )
+
+def detekt_test_suite(name):
+    _test_detekt_action_contents()
```

Comment:
Actually I wanted to write another test on missing `srcs` but... we execute fine without it. I guess we should throw an error on missing `srcs` or something.